lib/hook.js

```diff
@@ -174,6 +175,39 @@ function unhookRunInThisContext() {
     vm.runInThisContext = originalRunInThisContext;
 }
 
+/**
+ * hooks `vm.runInNewContext` to return transformed code.
+ * @method hookRunInNewContext
+ * @static
+ * @param matcher {Function(filePath)} a function that is called with the filename passed to `vm.runInNewContext`
+ *  Should return a truthy value when transformations need to be applied to the code, a falsy value otherwise
+ * @param transformer {Function(code, filePath)} a function called with the original code and the filename passed to
+ *  `vm.createScript`. Should return the transformed code.
+ * @param options {Object} options Optional.
+ * @param {Boolean} [options.verbose] write a line to standard error every time the transformer is called
+ */
+function hookRunInNewContext(matcher, transformer, opts) {
+    opts = opts || {};
+    var fn = transformFn(matcher, transformer, opts.verbose);
+    vm.runInNewContext = function (code, context, file) {
+        var ret = fn(code, file);
+        Object.keys(global).forEach(function(key) {
+          if (key.indexOf('$$cov_') === 0) {
```

Comment:
The `hook` library is completely agnostic of the fact that it is used in a coverage context and I would like to keep it that way. Could you change this as follows:

* Accept an `injectScope` object in hook options that has all the key values that need to be transferred to the new scope. When this is present copy all the keys down.
* Allow `opts` to be optional and the `injectScope` key to be missing and do nothing if that is the case
* Change the caller - which knows the coverage variable name - to add the injectScope object with the coverage variable name and contents. This also gets rid of the assumption you are making re: naming of the coverage object.